src/memray/_memray/tracking_api.cpp

```diff
@@ -885,14 +872,11 @@ Tracker::updateModuleCacheImpl()
     if (!d_unwind_native_frames) {
         return;
     }
-    auto writer_lock = d_writer->acquireLock();
-    if (!d_writer->writeRecordUnsafe(MemoryMapStart{})) {
-        std::cerr << "memray: Failed to write output, deactivating tracking" << std::endl;
-        deactivate();
-    }
+
+    std::vector<ImageSegments> mappings;
```

Comment:
We could use the value of `d_mappings_by_generation[-1].size()`

Reply:
Sure, we could, but I mean, 🤷 - this isn't a hot path.

But OK, I'll add a `.reserve()`

Reply:
... but I can't base it on `d_mappings_by_generation[-1].size()`, because that's an attribute of the `AggregatingRecordWriter`, and we're in `tracking_api.cpp`, which doesn't even know which concrete subclass of `RecordWriter` we've got.

I'll add a `static` variable to the function to track the last size.